src/storage/FileResourceStore.ts

```diff
@@ -0,0 +1,427 @@
+ */
+export class FileResourceStore implements ResourceStore {
+  private readonly baseRequestURI: string;
+  private readonly rootFilepath: string;
+  private readonly interactionController: InteractionController;
+
+  private readonly predicates = {
+    aType: DataFactory.namedNode(`${RDF}type`),
+    modified: DataFactory.namedNode(`${TERMS}modified`),
+    contains: DataFactory.namedNode(`${LDP}contains`),
+    mtime: DataFactory.namedNode(`${STAT}mtime`),
+    size: DataFactory.namedNode(`${STAT}size`),
+  };
+
+  private readonly objects = {
+    container: DataFactory.namedNode(`${LDP}Container`),
+    basicContainer: DataFactory.namedNode(`${LDP}BasicContainer`),
+    ldpResource: DataFactory.namedNode(`${LDP}Resource`),
+    dateTime: DataFactory.namedNode(`${XML}dateTime`),
+  };
+
+  /**
+   * @param baseRequestURI - Will be stripped of all incoming URIs and added to all outgoing ones to find the relative
+   * path.
+   * @param rootFilepath - Root filepath in which the resources and containers will be saved as files and directories.
+   * @param interactionController - Instance of InteractionController to use.
+   */
+  public constructor(baseRequestURI: string, rootFilepath: string, interactionController: InteractionController) {
+    this.baseRequestURI = trimTrailingSlashes(baseRequestURI);
+    this.rootFilepath = trimTrailingSlashes(rootFilepath);
+    this.interactionController = interactionController;
+  }
+
+  /**
+   * Store the incoming data as a file under a file path corresponding to `container.path`,
+   * where slashes correspond to subdirectories.
+   * @param container - The identifier to store the new data under.
+   * @param representation - Data to store. Only File streams are supported.
+   *
+   * @returns The newly generated identifier.
+   */
+  public async addResource(container: ResourceIdentifier, representation: Representation): Promise<ResourceIdentifier> {
+    if (representation.dataType !== DATA_TYPE_BINARY) {
+      throw new UnsupportedMediaTypeHttpError('FileResourceStore only supports binary representations.');
+    }
+    const path = this.parseIdentifier(container);
+    const { slug, raw } = representation.metadata;
+    const linkTypes = representation.metadata.linkRel?.type;
+    let metadata;
+    if (raw.length > 0) {
+      metadata = streamifyArray(raw).pipe(new StreamWriter({ format: 'text/turtle' }));
+    }
+
+    const isContainer = this.interactionController.isContainer(slug, linkTypes);
+    const newIdentifier = this.interactionController.generateIdentifier(isContainer, slug);
+    if (!isContainer) {
+      // Create a file for the resource with as filepath the parent container.
+      return this.createFile(path, newIdentifier, representation.data, path.endsWith('/'), metadata);
+    }
+
+    // Create a new container as subdirectory of the parent container.
+    return this.createContainer(path, newIdentifier, path.endsWith('/'), metadata);
```

Comment:
Right now, the first path appears to be a premature escape, whereas they are actually equivalent paths. This slight variation might show better what's going on.


```suggestion
    return isContainer ?
     // Create a new container as subdirectory of the parent container
     this.createContainer(path, newIdentifier, path.endsWith('/'), metadata)
    // Create a file for the resource with as filepath the parent container
    this.createFile(path, newIdentifier, representation.data, path.endsWith('/'), metadata);
```

Or just an `else` works too of course.

Reply:
I first wrote it with an else but then I got `ESLint: Unnecessary 'else' after 'return'(no-else-return)`.
I've changed it to your suggestion to make it clearer.